src/errors.rs

```diff
@@ -0,0 +1,73 @@
+use crate::de;
+use crate::ser;
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+/// Error type returned from the library
+pub enum AvroError {
+    /// Represents all cases of `std::io::Error`
+    #[error(transparent)]
+    IOError(#[from] std::io::Error),
```

Comment:
At my company we've been adopting a slightly different convention for naming error variants, and that naming convention is simply to remove the redundant `Error` suffix on the variants. It may seem awkward at first, but after awhile it grew on me, and now I prefer this way of writing errors. Since you've got a few catch-all variants in here, it'll look weird, but after converting the variants to reflect specific failure modes it's a lot better. So you end up with error constructions like: `Error::Decode`, `Error::ParseSchema`, etc.

Reply:
I don't consider this a blocker for landing the PR.

Reply:
Love the idea, I'll absolutely adapt that for my stuff!

Reply:
Ah, I think this is in line with the `thiserror` documentation, too. I kinda copied most of the error names (made bulk of changes easier), but I like this approach better!